build_early.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
```

Comment:
Maybe add 

```bash
# exit running on nonzero exit
set -o errexit
# check and abort on unbound vars
set -o nounset
# show errors in pipes
set -o pipefail
```

I am unsure about `errexit`, but I assume the commands return `0` on success.

Reply:
can I add these to a source file?

Reply:
This should work